java/vector/src/main/java/org/apache/arrow/vector/compare/VectorValueEqualizer.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.compare;
+
+import org.apache.arrow.vector.ValueVector;
+
+/**
+ * A function to determine if two vectors are equal at specified positions.
+ * @param <V> the vector type.
+ */
+public interface VectorValueEqualizer<V extends ValueVector> {
```

Comment:
why the generic?

Reply:
I think this is a general question. 
The value of generic is that it gives the Java compiler a chance to discover potential (type related) errors in the code, and prevent them from ever happening at runtime.